### ISSUE TITLE
Is there a way to know which rule has thrown the default 'Not Authorised' error?

### ISSUE THREAD

**Issue by faurehu** [maticzav/graphql-shield#231]
Title pretty much says it all.

**Comment by maticzav** [comment]
Hey @faurehu 👋,

I am not sure whether the title indeed says it all. Typically, `graphql` provides a path to the field/type where the error occurred. In that sense, yes, there's a way of telling which rule returned the default, `Not Authorised` error - the one assigned to that particular field.

On the other hand, `shield` provides you with the concept of `logic` rules. In that regard, it's not possible to know which rule threw an unauthorised error. However, you can customise the returned error to each specific rule by using [custom errors](https://github.com/maticzav/graphql-shield#custom-errors).

Furthermore, you could use `info` argument to the rule to determine the location of executing-rule during runtime.

I hope I gave you an idea of the breadth of this question, let me know if any of the above solved your problem! 🙂

**Comment by faurehu** [comment]
I already figured out what was going on, my problem was that I wasn't aware of what exactly was being called. For example I had a mutation `myMutation`  that returns a `myObject` with a field of type `myField`.

My permissions looked something like this: 
```
const permissions = shield({
  Mutation: {
    myMutation: allow
  },
  myObject: deny,
  myField: allow
})
```

I glossed over the fact that the mutation returns `myObject` so I'd need permission for that. And I thought because inside `myObject` there is a `myField`, I'd need to allow that. Anyway, after playing with multiple configurations I realised that I had to allow `myObject` too.

Or that's what I thought because I just tried a permission schema where everything but the mutation is allowed and it is working just fine. Now I'm more confused.

So in a nutshell now my question has turned into one about the delegation of permissions. If a mutation is allowed, does this permission waterfall into the return type and the types in its fields?

Thanks for your help btw.

**Comment by maticzav** [comment]
Hmm, that's an interesting question! Shield provides no delegation of `rules` however the intended behaviour of GraphQL's query execution might portray seemingly delegating functionality. I suggest you read an article on the topic of resolver execution that I've attached below. I believe it should help you greatly with the understanding of how `shield` and GraphQL in general work.

Let me know if it helped. 🙂 

https://www.prisma.io/blog/graphql-server-basics-the-schema-ac5e2950214e
